File: differ/src/main/java/ru/starksoft/differ/Logger.java
package ru.starksoft.differ;

import androidx.annotation.NonNull;

public interface Logger {

    void log(@NonNull Throwable t);

    void log(@NonNull String tag, @NonNull String message);

    void log(@NonNull String tag, @NonNull String message, @NonNull Throwable t);

    void w(@NonNull String tag, @NonNull String message);

    void d(@NonNull String tag, @NonNull String message);
}

File: diff-adapter/src/main/java/ru/starksoft/differ/Logger.kt
package ru.starksoft.differ

interface Logger {

	fun log(t: Throwable)

	fun log(tag: String, message: String)

	fun log(tag: String, message: String, t: Throwable)

	fun w(tag: String, message: String)

	fun d(tag: String, message: String)

	fun e(tag: String, message: String, t: Throwable)
}
